src/clients/http.rs

```diff
@@ -372,11 +358,10 @@ mod tests {
 
     #[tokio::test(flavor = "multi_thread")]
     async fn http_run_command_should_include_name() {
-        let image = GenericImage::new("hello-world");
         let docker = Http::new();
-
-        let run_args = RunArgs::default().with_name("hello_container");
-        let container = docker.run_with_args(image, run_args).await;
+        let image = GenericImage::new("hello-world");
+        let runnable = RunnableImage::from(image).with_name("hello_container");
```

Comment:
What do you think of an extension trait to `Image` that allows one to call `with_name` and all of the other functions directly on `Image`?

It would enable an API such as:

```rust
let container = docker.run(GenericImage::new("hello-world").with_name("hello_container")).await;
```

Reply:
We can ship this as a follow-up feature though, not a requirement for me to merge this.

Reply:
I did see `ImageExt` on your big PR. But wasn't sure why would we really need it. Let's discuss more on that.

Reply:
> Let's discuss more on that. I guess to avoid users to directly deal with `RunnableImage` ?

Exactly. In particular, the extension trait design achieves the following:

- It "hides" the conversion of `Image` -> `RunnableImage` from the user, making it seem like we are passing instances of `Image` to `run` in all circumstances.
- Contrary to adding default functions on `Image`, functionality defined via the extension trait cannot be overridden. (Overriding default functionality could break certain assumptions and hence broken functionality).
- It automatically applies to all `Image`s, reducing boilerplate for users and at the same time providing a rich interface.

The only downside is that users have to import another trait for this to work but I think that is acceptable :)